From 8db1d31e6715dce9d34dbcaaa85b38646b2ec3d1 Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Thu, 5 Mar 2020 09:48:44 +0800
Subject: [PATCH] initscripts:adapt to the grep command

Signed-off-by: zhanglu <zhanglu37@huawei.com>
---
 etc/rc.d/init.d/network | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/etc/rc.d/init.d/network b/etc/rc.d/init.d/network
index f69948e..9a0c3f3 100755
--- a/etc/rc.d/init.d/network
+++ b/etc/rc.d/init.d/network
@@ -132,7 +132,7 @@ start)
             continue
         fi
 
-        if LANG=C grep -EL "^ONBOOT=['\"]?[Nn][Oo]['\"]?" ifcfg-$i > /dev/null ; then
+        if ! LANG=C grep -EL "^ONBOOT=['\"]?[Nn][Oo]['\"]?" ifcfg-$i > /dev/null ; then
             # this loads the module, to preserve ordering
             is_available $i
             continue
@@ -149,7 +149,7 @@ start)
 
     # Bring up xDSL and VPN interfaces
     for i in $vlaninterfaces $bridgeinterfaces $xdslinterfaces $vpninterfaces ; do
-        if ! LANG=C grep -EL "^ONBOOT=['\"]?[Nn][Oo]['\"]?" ifcfg-$i >/dev/null 2>&1 ; then
+        if LANG=C grep -EL "^ONBOOT=['\"]?[Nn][Oo]['\"]?" ifcfg-$i >/dev/null 2>&1 ; then
             if [ "$godamon" != "1" ] ; then
 		 action $"Bringing up interface $i: " ./ifup $i boot
 		 [ $? -ne 0 ] && rc=1
-- 
1.8.3.1

